### PR TITLE
[fix](nereids) fix bugs in nereids window function

### PR DESCRIPTION
**Pull request by englefly**
# Proposed changes
support tpcds related window function in nereids
## NormalizeAggregate rule: 
- push agg-fun in windowExpression down to AggregateNode
for example:
`select sum(sum(a)) over (order by b)`
plan
windowExpression( sum(y) over (order by b))
     +---->Agg(sum(a) as y, b)

- push other expr to upper proj
`select sum(a+1) over ()`
plan
windowExpression(sum(y) over ())
   +--->proj(a + 1 as y,...)
             +--->Agg(a,...)

## ExtractAndNormalizeWindowExpression rule
old version ignored expressions which contains windExpression. 
for example: 'avg(a) / sum(sum(a)) over()', avg(a) is ignored.

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]
LGTM

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/ExtractAndNormalizeWindowExpression.java [hunk 1]
```diff
@@ -61,7 +65,8 @@ public Rule build() {
             } else {
                 boolean needAggregate = bottomProjects.stream().anyMatch(expr ->
                         expr.anyMatch(AggregateFunction.class::isInstance));
-                if (needAggregate) {
+                boolean hasAggChild = isProjectOnAggregate(project);
```
[the remark this answers is not in the document]
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/ExtractAndNormalizeWindowExpression.java]: "isProjectOnAggregate" can be removed.
If an agg-fun need to be push down and the child is AggNode, the agg-fun is already pushed down in NormalizeAggregation Rule.

**Comment by englefly** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**